api/src/services/hashnode.ts

```diff
@@ -1,13 +1,36 @@
 import { request } from 'graphql-request'
-import gql from 'graphql-tag'
+import type { Post, Publication, QueryrecentPostsArgs } from 'types/graphql'
 
-export const recentPosts = async () => {
-  const query = gql`
+import { logger } from 'src/lib/logger'
+
+type PublicationResponse = {
+  isTeam: boolean
+  title: string
+  posts: {
+    edges: Array<{
+      node: Post
+    }>
+  }
+}
+
+type RecentPostsResponse = {
+  publication: PublicationResponse
+}
+/**
+ * Fetches a Publication with latest posts from Hashnode.
+ *
+ * Note: This resolver is cached for a specified amount of time
+ * defined in the GraphQL Handler in ttlPerSchemaCoordinate.
+ **/
+export const recentPosts = async ({
+  first = 3,
+}: QueryrecentPostsArgs): Promise<Publication> => {
+  const RECENT_POSTS = `
```

Comment:
Just a string, not a gel template so should not encounter limiting issues when gel thinks it checks against app schema vs hash node schema